aviary/wrenformer/run.py

```diff
@@ -0,0 +1,334 @@
+from __future__ import annotations
+
+import os
+from typing import Any, Literal, Sequence
+
+import numpy as np
+import pandas as pd
+import torch
+import wandb
+from torch import nn
+from torch.optim.swa_utils import SWALR, AveragedModel
+
+from aviary import ROOT
+from aviary.core import Normalizer, TaskType
+from aviary.data import InMemoryDataLoader
+from aviary.losses import RobustL1Loss
+from aviary.utils import get_metrics
+from aviary.wrenformer.data import (
+    collate_batch,
+    get_composition_embedding,
+    wyckoff_embedding_from_aflow_str,
+)
+from aviary.wrenformer.model import Wrenformer
+from aviary.wrenformer.utils import print_walltime
+
+__author__ = "Janosh Riebesell"
+__date__ = "2022-06-12"
+
+torch.manual_seed(0)  # ensure reproducible results
+
+reg_key, clf_key = "regression", "classification"
+
+
+@print_walltime(end_desc="run_wrenformer()")
+def run_wrenformer(
+    run_name: str,
+    task_type: TaskType,
+    timestamp: str,
+    train_df: pd.DataFrame,
+    test_df: pd.DataFrame,
+    target_col: str,
+    epochs: int,
+    id_col: str = "material_id",
+    n_attn_layers: int = 4,
+    wandb_project: str = None,
+    checkpoint: Literal["local", "wandb"] | None = None,
+    swa_start=0.7,  # start SWA after 50% of epochs
```

Comment:
70% in note

Reply:
Will remove the comment. Doc string explains it better anyway.